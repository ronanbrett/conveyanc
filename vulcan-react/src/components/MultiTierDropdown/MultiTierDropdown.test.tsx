import React from "react";
import { render } from "@testing-library/react";
import {
  MultiTierDropdown,
  MultiTierDropdownOption,
  MultiTierDropdownItem,
} from "./index";
import { Field, Form, Formik } from "formik";
import { create } from "react-test-renderer";

const TestComponent = () => {
  return (
    <Formik initialValues={{ propertyType: "" }} onSubmit={(x) => console.log(x)}>
      <Form>
        <label htmlFor="propertyType">Property Type</label>
        <Field name="propertyType">
        {() => (
          <MultiTierDropdown name="propertyType" placeholder="Select Property Type">
            <MultiTierDropdownItem value="Group2" label="test">
              <MultiTierDropdownOption value="TestA">Test</MultiTierDropdownOption>
            </MultiTierDropdownItem>
            <MultiTierDropdownItem value="Group1" label="test">
              <MultiTierDropdownOption value="TestB">Test</MultiTierDropdownOption>
              <MultiTierDropdownOption value="TestC">Test</MultiTierDropdownOption>
              <MultiTierDropdownOption value="TestD">Test</MultiTierDropdownOption>
            </MultiTierDropdownItem>
            <MultiTierDropdownItem value="Group3" label="test">
              <MultiTierDropdownOption value="TestE">Test</MultiTierDropdownOption>
              <MultiTierDropdownOption value="TestF">Test</MultiTierDropdownOption>
            </MultiTierDropdownItem>
          </MultiTierDropdown>
        )}
        </Field>
      </Form>
    </Formik>
  );
};

describe("MultiTierDropdown", () => {
  it("renders without error", () => {
    const component = create(<TestComponent />);
    expect(component).toMatchSnapshot();
  });
});
