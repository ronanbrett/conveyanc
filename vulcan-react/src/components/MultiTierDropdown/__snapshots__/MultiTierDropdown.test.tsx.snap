// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiTierDropdown renders without error 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <label
    htmlFor="propertyType"
  >
    Property Type
  </label>
  <div
    className="multi-dd field--valid"
  >
    <header
      className="multi-dd__trigger"
      onClick={[Function]}
    >
      <div
        className="multi-dd__trigger-content"
      >
        Select Property Type
      </div>
      <button
        className="icon-button icon-button--default mulit-dd__trigger-btn"
        type="button"
      >
        <div
          className="icon__container"
        >
          <i
            className="material-icons"
          >
            keyboard_arrow_down
          </i>
          <div
            className="icon__background"
          />
        </div>
      </button>
    </header>
    <div
      className="multi-dd__content hidden"
    >
      <div
        className="multi-dd__item"
      >
        <header
          className="multi-dd__item-trigger"
          onClick={[Function]}
        >
          <h1
            className="multi-dd__item-trigger-content"
          >
            test
          </h1>
          <button
            className="icon-button icon-button--mini multi-dd__trigger-btn"
            type="button"
          >
            <div
              className="icon__container"
            >
              <i
                className="material-icons"
              >
                keyboard_arrow_down
              </i>
              <div
                className="icon__background"
              />
            </div>
          </button>
        </header>
        <ul
          className="multi-dd__option-content hidden"
          onKeyDown={[Function]}
          role="listbox"
          tabIndex={-1}
        >
          <li
            aria-selected={false}
            className="multi-dd__option-list-item  "
            onClick={[Function]}
            role="option"
          >
            Test
          </li>
        </ul>
      </div>
      <div
        className="multi-dd__item"
      >
        <header
          className="multi-dd__item-trigger"
          onClick={[Function]}
        >
          <h1
            className="multi-dd__item-trigger-content"
          >
            test
          </h1>
          <button
            className="icon-button icon-button--mini multi-dd__trigger-btn"
            type="button"
          >
            <div
              className="icon__container"
            >
              <i
                className="material-icons"
              >
                keyboard_arrow_down
              </i>
              <div
                className="icon__background"
              />
            </div>
          </button>
        </header>
        <ul
          className="multi-dd__option-content hidden"
          onKeyDown={[Function]}
          role="listbox"
          tabIndex={-1}
        >
          <li
            aria-selected={false}
            className="multi-dd__option-list-item  "
            onClick={[Function]}
            role="option"
          >
            Test
          </li>
          <li
            aria-selected={false}
            className="multi-dd__option-list-item  "
            onClick={[Function]}
            role="option"
          >
            Test
          </li>
          <li
            aria-selected={false}
            className="multi-dd__option-list-item  "
            onClick={[Function]}
            role="option"
          >
            Test
          </li>
        </ul>
      </div>
      <div
        className="multi-dd__item"
      >
        <header
          className="multi-dd__item-trigger"
          onClick={[Function]}
        >
          <h1
            className="multi-dd__item-trigger-content"
          >
            test
          </h1>
          <button
            className="icon-button icon-button--mini multi-dd__trigger-btn"
            type="button"
          >
            <div
              className="icon__container"
            >
              <i
                className="material-icons"
              >
                keyboard_arrow_down
              </i>
              <div
                className="icon__background"
              />
            </div>
          </button>
        </header>
        <ul
          className="multi-dd__option-content hidden"
          onKeyDown={[Function]}
          role="listbox"
          tabIndex={-1}
        >
          <li
            aria-selected={false}
            className="multi-dd__option-list-item  "
            onClick={[Function]}
            role="option"
          >
            Test
          </li>
          <li
            aria-selected={false}
            className="multi-dd__option-list-item  "
            onClick={[Function]}
            role="option"
          >
            Test
          </li>
        </ul>
      </div>
    </div>
  </div>
</form>
`;
